Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -304,11 +304,16 @@ public typealias StartPurchaseBlock = (@escaping PurchaseCompletedBlock) -> Void
         let receiptRefreshRequestFactory = ReceiptRefreshRequestFactory()
         let fetcher = StoreKitRequestFetcher(requestFactory: receiptRefreshRequestFactory,
                                              operationDispatcher: operationDispatcher)
+
+        let apiKeyValidationResult = Configuration.validate(apiKey: apiKey)
+        apiKeyValidationResult.logIfNeeded()
```

Comment:
Since we now need the API key validation result in the `init` of the `Purchases` instance, I thought it would make more sense to add the log here. From the developers perspective, it should be seamless. Lmk if you think otherwise

Reply:
Hmm right... I guess it's not entirely seamless since now we will only log once the SDK is configured, not when creating the `Configuration` object right? 

I wonder if we should instead store the verification result in the `Configuration` object as an `internal` property and access it here when initializing purchases... Not sure how much of a lift this would be though, so not a strong opinion.

Reply:
Yeah I'm honestly ok with this change :) 99% of the time, you would only create this object to configure the SDK, so it would be pretty much seamless 👍

Reply:
Btw, I did a small refactor in https://github.com/RevenueCat/purchases-ios/pull/5386/commits/a2609f3ec34c265952556556c7fd205ef983b0a2 to make the implementation similar to what you've done in Android, which is cleaner